validator/src/main/java/com/google/daq/mqtt/sequencer/sequences/DiscoverySequences.java

```diff
@@ -182,23 +182,24 @@ private boolean shouldEnumerate(Depth depth) {
   private void checkFeatureDiscovery(Map<String, FeatureDiscovery> features) {
     Set<String> enumeratedFeatures = features.entrySet().stream()
         .filter(DiscoverySequences::isActive).map(Entry::getKey).collect(Collectors.toSet());
-    requireNonNull(deviceMetadata.features, "device metadata features missing");
-    Set<String> enabledFeatures = deviceMetadata.features.entrySet().stream()
-        .filter(DiscoverySequences::isActive).map(Entry::getKey).collect(Collectors.toSet());
-    SetView<String> extraFeatures = Sets.difference(enumeratedFeatures, enabledFeatures);
-    SetView<String> missingFeatures = Sets.difference(enabledFeatures, enumeratedFeatures);
-    SetView<String> difference = Sets.union(extraFeatures, missingFeatures);
-    String details = format("missing { %s }, extra { %s }", CSV_JOINER.join(missingFeatures),
-        CSV_JOINER.join(extraFeatures));
-    checkThat("feature enumeration matches metadata", difference::isEmpty, details);
+    checkFeatureMetadata(enumeratedFeatures);
     Set<String> unofficial = enumeratedFeatures.stream()
         .filter(feature -> !Bucket.contains(feature)).collect(Collectors.toSet());
     String format = format("unrecognized { %s }", CSV_JOINER.join(unofficial));
     checkThat("all enumerated features are official buckets", unofficial::isEmpty, format);
   }
 
-  private boolean isTrue(Boolean condition) {
-    return ofNullable(condition).orElse(false);
+  private void checkFeatureMetadata(Set<String> enumeratedFeatures) {
+    ifNotNullThen(deviceMetadata.features, metadataFeatures -> {
```

Comment:
I think this should be `ifNullSkipTest` -- if metadata is not defined, then it's not that the test passes, it's that it shouldn't be tested at all...